swarm.sh

```diff
@@ -1,94 +1,148 @@
-    
-    log info "Restarting HAPI FHIR.."
-    try "docker service scale instant_hapi-fhir=0" "Error scaling down hapi-fhir to update the IG"
-    try "docker service scale instant_hapi-fhir=$HAPI_FHIR_INSTANCES" "Error scaling up hapi-fhir to update the IG"
-
-    log info "Removing stale configs..."
-    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "superset"
-    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "clickhouse"
-    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "hapi-fhir"
-
-    log info "Cleaning containers and configs"
+  log info "Removing stale configs..."
+  for config_label in "${CONFIG_LABELS[@]}"; do
+    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "$config_label"
+  done
+}
+
+function initialize_package() {
+  if [[ "${ACTION}" == "up" ]]; then
     clean_containers_and_configs
+  fi
+
+  config::set_config_digests "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml
+
+  # Clickhouse
+  config::generate_service_configs cares-clickhouse-config-importer / "${COMPOSE_FILE_PATH}/importer/analytics-datastore-clickhouse/" "${COMPOSE_FILE_PATH}/importer" clickhouse
+  clickhouse_temp_compose_param="-c ${COMPOSE_FILE_PATH}/importer/docker-compose.tmp.yml"
+
+  try "docker stack deploy -c ${COMPOSE_FILE_PATH}/importer/docker-compose.config.yml $clickhouse_temp_compose_param instant" throw "Failed to deploy Cares on Platform"
+
+  log info "Waiting to update configs"
+  # Kafka Mapper Consumer
+  REF_service_update_args=""
+  config::update_service_configs REF_service_update_args /app/src/data "$COMPOSE_FILE_PATH"/importer/kafka-mapper-consumer/mapping kafka-mapper-consumer
+  config::update_service_configs REF_service_update_args /app/src/plugin "$COMPOSE_FILE_PATH"/importer/kafka-mapper-consumer/plugin kafka-mapper-consumer
+  try "docker service update $REF_service_update_args instant_kafka-mapper-consumer" throw "Failed to update config for instant_kafka-mapper-consumer"
+  # Superset
+  REF_service_update_args=""
+  config::update_service_configs REF_service_update_args /app/pythonpath "$COMPOSE_FILE_PATH"/importer/dashboard-visualiser-superset superset
+  # TODO: Update .env.superset once the value for MAPBOX_API_KEY is known
+  config::env_var_add_from_file REF_service_update_args "$COMPOSE_FILE_PATH"/.env.superset
+  try "docker service update $REF_service_update_args instant_dashboard-visualiser-superset" throw "Failed to update config for instant_dashboard-visualiser-superset"
+
+  cleaning
 
-    # Exit 0 here to avoid false error messages
-    exit 0
+  restart_hapi_fhir
+
+  clean_containers_and_configs
+}
+
+main() {
+  init_vars "$@"
+  import_sources
+
+  if [[ "${ACTION}" == "init" ]] || [[ "${ACTION}" == "up" ]]; then
+    log info "Running package in single node mode"
```

Comment:
Is this package only ever run in single mode?

Reply:
hmm good point ... fixed it